File: dataset/ZeroShot - Copy/OPENAI_gpt-5-2025-08-07/Task150.cpp
#include <iostream>
#include <vector>
#include <new>

class SinglyLinkedList {
private:
    struct Node {
        int val;
        Node* next;
        explicit Node(int v) : val(v), next(nullptr) {}
    };
    Node* head;
    Node* tail;
    std::size_t sz;

public:
    SinglyLinkedList() : head(nullptr), tail(nullptr), sz(0) {}

    bool insert(int value) {
        Node* n = new (std::nothrow) Node(value);
        if (n == nullptr) {
            return false;
        }
        if (head == nullptr) {
            head = n;
            tail = n;
        } else {
            tail->next = n;
            tail = n;
        }
        ++sz;
        return true;
    }

    bool deleteValue(int value) {
        Node* prev = nullptr;
        Node* curr = head;
        while (curr != nullptr) {
            if (curr->val == value) {
                if (prev == nullptr) {
                    head = curr->next;
                } else {
                    prev->next = curr->next;
                }
                if (curr == tail) {
                    tail = prev;
                }
                delete curr;
                --sz;
                return true;
            }
            prev = curr;
            curr = curr->next;
        }
        return false;
    }

    bool search(int value) const {
        Node* curr = head;
        while (curr != nullptr) {
            if (curr->val == value) return true;
            curr = curr->next;
        }
        return false;
    }

    std::size_t size() const { return sz; }

    std::vector<int> toVector() const {
        std::vector<int> out;
        out.reserve(sz);
        Node* curr = head;
        while (curr != nullptr) {
            out.push_back(curr->val);
            curr = curr->next;
        }
        return out;
    }

    ~SinglyLinkedList() {
        Node* curr = head;
        while (curr != nullptr) {
            Node* next = curr->next;
            delete curr;
            curr = next;
        }
        head = nullptr;
        tail = nullptr;
        sz = 0;
    }
};

static void printVec(const std::vector<int>& v) {
    std::cout << "[";
    for (std::size_t i = 0; i < v.size(); ++i) {
        if (i) std::cout << ", ";
        std::cout << v[i];
    }
    std::cout << "]\n";
}

int main() {
    // Test Case 1
    SinglyLinkedList l1;
    l1.insert(1); l1.insert(2); l1.insert(3);
    std::cout << "TC1 search 2: " << (l1.search(2) ? "true" : "false") << "\n";
    std::cout << "TC1 delete 2: " << (l1.deleteValue(2) ? "true" : "false") << "\n";
    std::cout << "TC1 search 2: " << (l1.search(2) ? "true" : "false") << "\n";
    std::cout << "TC1 list: "; printVec(l1.toVector());

    // Test Case 2: delete head
    SinglyLinkedList l2;
    l2.insert(10); l2.insert(20);
    std::cout << "TC2 delete 10: " << (l2.deleteValue(10) ? "true" : "false") << "\n";
    std::cout << "TC2 list: "; printVec(l2.toVector());

    // Test Case 3: delete tail
    SinglyLinkedList l3;
    l3.insert(7); l3.insert(8); l3.insert(9);
    std::cout << "TC3 delete 9: " << (l3.deleteValue(9) ? "true" : "false") << "\n";
    std::cout << "TC3 list: "; printVec(l3.toVector());

    // Test Case 4: delete non-existent
    SinglyLinkedList l4;
    l4.insert(5);
    std::cout << "TC4 delete 6: " << (l4.deleteValue(6) ? "true" : "false") << "\n";
    std::cout << "TC4 list: "; printVec(l4.toVector());

    // Test Case 5: search empty
    SinglyLinkedList l5;
    std::cout << "TC5 search 42: " << (l5.search(42) ? "true" : "false") << "\n";
    std::cout << "TC5 list: "; printVec(l5.toVector());

    return 0;
}

File: dataset/ZeroShot/OPENAI_gpt-5-2025-08-07/Task150.java
import java.util.Arrays;

public class Task150 {
    private static final class Node {
        final int val;
        Node next;
        Node(int v) { this.val = v; this.next = null; }
    }

    private Node head;
    private Node tail;
    private int size;

    public Task150() {
        this.head = null;
        this.tail = null;
        this.size = 0;
    }

    // Insert at tail to preserve insertion order
    public boolean insert(int value) {
        Node n = new Node(value);
        if (head == null) {
            head = n;
            tail = n;
        } else {
            tail.next = n;
            tail = n;
        }
        size++;
        return true;
    }

    // Delete first occurrence of value
    public boolean delete(int value) {
        Node prev = null;
        Node curr = head;
        while (curr != null) {
            if (curr.val == value) {
                if (prev == null) {
                    head = curr.next;
                } else {
                    prev.next = curr.next;
                }
                if (curr == tail) {
                    tail = prev;
                }
                curr.next = null; // help GC
                size--;
                return true;
            }
            prev = curr;
            curr = curr.next;
        }
        return false;
    }

    public boolean search(int value) {
        Node curr = head;
        while (curr != null) {
            if (curr.val == value) return true;
            curr = curr.next;
        }
        return false;
    }

    public int size() {
        return size;
    }

    public int[] toArray() {
        int[] arr = new int[size];
        int i = 0;
        Node curr = head;
        while (curr != null) {
            arr[i++] = curr.val;
            curr = curr.next;
        }
        return arr;
    }

    private static void printArray(int[] arr) {
        System.out.println(Arrays.toString(arr));
    }

    public static void main(String[] args) {
        // Test Case 1
        Task150 list1 = new Task150();
        list1.insert(1);
        list1.insert(2);
        list1.insert(3);
        System.out.println("TC1 search 2: " + list1.search(2));
        System.out.println("TC1 delete 2: " + list1.delete(2));
        System.out.println("TC1 search 2: " + list1.search(2));
        System.out.print("TC1 list: ");
        printArray(list1.toArray());

        // Test Case 2: delete head
        Task150 list2 = new Task150();
        list2.insert(10);
        list2.insert(20);
        System.out.println("TC2 delete 10: " + list2.delete(10));
        System.out.print("TC2 list: ");
        printArray(list2.toArray());

        // Test Case 3: delete tail
        Task150 list3 = new Task150();
        list3.insert(7);
        list3.insert(8);
        list3.insert(9);
        System.out.println("TC3 delete 9: " + list3.delete(9));
        System.out.print("TC3 list: ");
        printArray(list3.toArray());

        // Test Case 4: delete non-existent
        Task150 list4 = new Task150();
        list4.insert(5);
        System.out.println("TC4 delete 6: " + list4.delete(6));
        System.out.print("TC4 list: ");
        printArray(list4.toArray());

        // Test Case 5: search in empty list
        Task150 list5 = new Task150();
        System.out.println("TC5 search 42: " + list5.search(42));
        System.out.print("TC5 list: ");
        printArray(list5.toArray());
    }
}